prompt-service/src/unstract/prompt_service/main.py

```diff
@@ -286,108 +285,121 @@ def prompt_processor() -> Any:
                 "Unable to obtain LLM / embedding / vectorDB",
             )
             return APIError(message=msg)
+        try:
+            vector_index = vector_db.get_vector_store_index()
 
-        vector_index = vector_db.get_vector_store_index()
-
-        context = ""
-        if output[PSKeys.CHUNK_SIZE] == 0:
-            # We can do this only for chunkless indexes
-            context: Optional[str] = tool_index.get_text_from_index(
-                embedding_instance_id=output[PSKeys.EMBEDDING],
-                vector_db_instance_id=output[PSKeys.VECTOR_DB],
-                doc_id=doc_id,
-                usage_kwargs=usage_kwargs,
-            )
-            if not context:
-                # UN-1288 For Pinecone, we are seeing an inconsistent case where
-                # query with doc_id fails even though indexing just happened.
-                # This causes the following retrieve to return no text.
-                # To rule out any lag on the Pinecone vector DB write,
-                # the following sleep is added.
-                # Note: This will not fix the issue. Since this issue is inconsistent
-                # and not reproducible easily, this is just a safety net.
-                time.sleep(2)
-                context: Optional[str] = tool_index.get_text_from_index(
+            context = ""
+            if output[PSKeys.CHUNK_SIZE] == 0:
+                # We can do this only for chunkless indexes
+                context: Optional[str] = index.query_text_from_index(
                     embedding_instance_id=output[PSKeys.EMBEDDING],
                     vector_db_instance_id=output[PSKeys.VECTOR_DB],
                     doc_id=doc_id,
                     usage_kwargs=usage_kwargs,
                 )
-                if context is None:
-                    # TODO: Obtain user set name for vector DB
-                    msg = "Couldn't fetch context from vector DB"
-                    app.logger.error(
-                        f"{msg} {output[PSKeys.VECTOR_DB]} for doc_id {doc_id}"
+                if not context:
+                    # UN-1288 For Pinecone, we are seeing an inconsistent case where
+                    # query with doc_id fails even though indexing just happened.
+                    # This causes the following retrieve to return no text.
+                    # To rule out any lag on the Pinecone vector DB write,
+                    # the following sleep is added.
+                    # Note: This will not fix the issue. Since this issue is
+                    # inconsistent, and not reproducible easily,
+                    # this is just a safety net.
+                    time.sleep(2)
```

Comment:
@gaya3-zipstack Shouldn't this be done for any chunk size and not just for zero chunk size?

Also from our side, if context not found error is still thrown on second try after 2 seconds, then we could add a suffix to the error message like:
`This happens usually due to a delay by the Vector DB provider to confirm writes to db. Please try again after some time.`

Reply:
Yes. For chunk size > 0, it is happening in https://github.com/Zipstack/unstract/blob/3f86e97724de405127d801a4a41354f36c5d1f3e/prompt-service/src/unstract/prompt_service/main.py#L704

However there was no check to see context was retrieved for non-zero chunk sizes before. I have added it now.
Updated error message as mentioned here.